source/common/grpc/common.cc

```diff
@@ -24,13 +24,63 @@ namespace Grpc {
 
 const std::string Common::GRPC_CONTENT_TYPE{"application/grpc"};
 
-void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& grpc_service,
-                        const std::string& grpc_method, bool success) {
+void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& protocol,
+                        const std::string& grpc_service, const std::string& grpc_method,
+                        const Http::HeaderEntry* grpc_status) {
+  if (!grpc_status) {
+    return;
+  }
+  uint64_t grpc_status_code;
+  bool success =
+      StringUtil::atoul(grpc_status->value().c_str(), grpc_status_code) && grpc_status_code == 0;
+  chargeStat(cluster, protocol, grpc_service, grpc_method, success);
+}
+
+void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& protocol,
+                        const std::string& grpc_service, const std::string& grpc_method,
+                        bool success) {
+  cluster.statsScope()
+      .counter(fmt::format("{}.{}.{}.{}", protocol, grpc_service, grpc_method,
+                           success ? "success" : "failure"))
+      .inc();
   cluster.statsScope()
-      .counter(
-           fmt::format("grpc.{}.{}.{}", grpc_service, grpc_method, success ? "success" : "failure"))
+      .counter(fmt::format("{}.{}.{}.total", protocol, grpc_service, grpc_method))
       .inc();
-  cluster.statsScope().counter(fmt::format("grpc.{}.{}.total", grpc_service, grpc_method)).inc();
+}
+
+void Common::chargeStat(const Upstream::ClusterInfo& cluster, const std::string& grpc_service,
+                        const std::string& grpc_method, bool success) {
+  chargeStat(cluster, "grpc", grpc_service, grpc_method, success);
```

Comment:
It's under Grpc namespace. I think that's why this method doesn't have Grpc in its name from beginning. The overloaded variant is used in other places when backend is Grpc.
https://github.com/lyft/envoy/blob/43c59df63b682c742ff586bd6ee78a648dd218ad/source/common/tracing/lightstep_tracer_impl.cc#L164
https://github.com/lyft/envoy/blob/34d308e9839d8c10e4ba5c06e323dd35ac6149c6/source/common/grpc/rpc_channel_impl.cc#L46